src/dbnode/integration/replace_node_test.go

```diff
@@ -0,0 +1,362 @@
+		for _, series := range expectedSeriesMaps[i] {
+			for _, elem := range series {
+				expectedSeriesIDs[i][elem.ID.String()] = struct{}{}
+			}
+		}
+	}
+	require.Equal(t, 2, len(expectedSeriesIDs[0]))
+	require.Equal(t, 1, len(expectedSeriesIDs[1]))
+
+	// Start the first server with filesystem bootstrapper.
+	require.NoError(t, setups[0].StartServer())
+
+	// Start the last server with peers and filesystem bootstrappers, no shards
+	// are assigned at first.
+	require.NoError(t, setups[1].StartServer())
+	log.Debug("servers are now up")
+
+	// Stop the servers on test completion.
+	defer func() {
+		setups.parallel(func(s TestSetup) {
+			require.NoError(t, s.StopServer())
+		})
+		log.Debug("servers are now down")
+	}()
+
+	// Bootstrap the new shards.
+	log.Debug("resharding to initialize shards on second node")
+	svc.SetInstances(instances.replacing)
+
+	svcs.NotifyServiceUpdate("m3db")
+	go func() {
+		for {
+			time.Sleep(time.Second)
+			for _, setup := range setups {
+				now = now.Add(time.Second)
+				setup.SetNowFn(now)
+			}
+		}
+	}()
+
+	// Generate some new data that will be written to the node while peer streaming is taking place
+	// to make sure that the data that is streamed in and the data that is received while streaming
+	// is going on are both handled correctly. In addition, this will ensure that we hold onto both
+	// sets of data durably after topology changes and that the node can be properly bootstrapped
+	// from just the filesystem and commitlog in a later portion of the test.
+	seriesToWriteDuringPeerStreaming := []string{
+		"series_after_bootstrap1",
+		"series_after_bootstrap2",
+	}
+	// Ensure that the new series belong that we're going to write belong to the host that is peer
+	// streaming data.
+	for _, seriesName := range seriesToWriteDuringPeerStreaming {
+		shard := shardSet.Lookup(ident.StringID(seriesName))
+		require.True(t, shard > midShard,
+			fmt.Sprintf("series: %s does not shard to second host", seriesName))
+	}
+	seriesReceivedDuringPeerStreaming := generate.BlocksByStart([]generate.BlockConfig{
+		{IDs: seriesToWriteDuringPeerStreaming, NumPoints: 90, Start: blockStart},
+	})
+	// Merge the newly generated series into the expected series map.
+	for blockStart, series := range seriesReceivedDuringPeerStreaming {
+		expectedSeriesMaps[1][blockStart] = append(expectedSeriesMaps[1][blockStart], series...)
+	}
+
+	// Spin up a background goroutine to issue the writes to the node while its streaming data
+	// from its peer.
+	doneWritingWhilePeerStreaming := make(chan struct{})
+	go func() {
+		for _, testData := range seriesReceivedDuringPeerStreaming {
+			err := setups[1].WriteBatch(namesp.ID(), testData)
+			if ShardsLeavingAndInitializingCountTowardsConsistency && err != nil {
+				// We don't expect any errors because even node is in leaving and init state but we have enabled the
+				// flag ShardsLeavingAndInitializingCountTowardsConsistency
+				panic(err)
```

Comment:
don't panic here. if its not something you're expecting - use t.Fail()/require/assert